src/testWithSpringBoot_2_1/java/org/openrewrite/java/spring/NoRepoAnnotationOnRepoSubclassTest.java

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright 2022 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+public class NoRepoAnnotationOnRepoSubclassTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new NoRepoAnnotationOnRepoSubclass())
+          .parser(JavaParser.fromJavaVersion().classpath("spring-context", "spring-beans", "spring-data"));
+    }
+
+    @Test
+    void simpleCase() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.springframework.stereotype.Repository;
+
+              @Repository
+              public class MyRepo implements org.springframework.data.repository.Repository {
+              }
+              """,
+            """
+
+              public class MyRepo implements org.springframework.data.repository.Repository {
+              }
+              """
+          )
+        );
+    }
+
+    @Test
```

Comment:
This is not a valid test, actually. The annotation must only be removed from repository *interfaces* extending a Spring Data repository, not manual implementations of such. Otherwise, the implementation types are not registered as Spring beans anymore.

Reply:
Aha, got it